src/main/go/properties/properties.go

```diff
@@ -0,0 +1,399 @@
+			// need check if escaped.
+			if (c == '=' || c == ':') && !precedingBackslash {
+				valueStart = keyLen + 1
+				hasSep = true
+				break
+			} else if (c == ' ' || c == '\t' || c == '\f') && !precedingBackslash {
+				valueStart = keyLen + 1
+				break
+			}
+			if c == '\\' {
+				precedingBackslash = !precedingBackslash
+			} else {
+				precedingBackslash = false
+			}
+			keyLen++
+		}
+		for _, c = range line[valueStart:] {
+			if c != ' ' && c != '\t' && c != '\f' {
+				if !hasSep && (c == '=' || c == ':') {
+					hasSep = true
+				} else {
+					break
+				}
+			}
+			valueStart++
+		}
+		key, err := loadConvert([]byte(line), 0, keyLen)
+		if err != nil {
+			return nil, err
+		}
+		value, err := loadConvert([]byte(line), valueStart, len(line)-valueStart)
+		if err != nil {
+			return nil, err
+		}
+		properties[key] = value
+	}
+	return properties, nil
+}
+
+func loadConvert(in []byte, off int, len int) (string, error) {
+	var aChar byte
+	end := off + len
+	start := off
+	for off < end {
+		aChar = in[off]
+		off++
+		if aChar == '\\' {
+			break
+		}
+	}
+	if off == end {
+		// No backslash
+		if len == 0 {
+			return "", nil
+		}
+		return string(in[start : start+len]), nil
+	}
+
+	// backslash found at off - 1, rewind offset
+	off--
+	out := []byte{}
+	if off-start != 0 {
+		out = in[start:off]
+	}
+
+	for off < end {
+		aChar = in[off]
+		off++
+		if aChar != '\\' {
+			out = append(out, aChar)
+			continue
+		}
+		// No need to bounds check since LineReader::readLine excludes
+		// unescaped \s at the end of the line
+		aChar = in[off]
+		off++
+		if aChar == 'u' {
+			// Read the xxxx
+			if off > end-4 {
+				return "", errors.New("malformed \\uxxxx encoding")
+			}
+			var value int
+			for i := 0; i < 4; i++ {
+				aChar = in[off]
+				off++
+				switch aChar {
+				case '0', '1', '2', '3', '4', '5', '6', '7', '8', '9':
+					value = (value << 4) + int(aChar) - '0'
+				case 'a', 'b', 'c', 'd', 'e', 'f':
+					value = (value << 4) + 10 + int(aChar) - 'a'
+				case 'A', 'B', 'C', 'D', 'E', 'F':
+					value = (value << 4) + 10 + int(aChar) - 'A'
+				default:
+					return "", errors.New("malformed \\uxxxx encoding")
+				}
+			}
+			out = utf8.AppendRune(out, rune(value))
+		} else {
+			if aChar == 't' {
+				aChar = '\t'
+			} else if aChar == 'r' {
+				aChar = '\r'
+			} else if aChar == 'n' {
+				aChar = '\n'
+			} else if aChar == 'f' {
+				aChar = '\f'
+			}
+			out = append(out, aChar)
+		}
+	}
+	return string(out), nil
+}
+
+func Resolve(fs afero.Fs, filename string, key string) (string, error) {
+	properties, err := LoadFile(fs, filename)
+	if err != nil {
+		return "", err
+	}
+
+	if value, ok := properties[key]; ok {
+		if decodedValue, err := decodeEnvProperty(value); err != nil {
+			return "", err
+		} else {
+			return decodedValue, nil
+		}
+	}
+
+	return "", fmt.Errorf("there is no property named %s in %s", key, filename)
+}
+
+func decodeEnvProperty(value string) (string, error) {
```

Comment:
Why are we doing this here? This is handled by Airlift's config processing support. In fact, the launcher shouldn't do any processing of application properties files.

Reply:
We've used that in other context for Starburst Enterprise. It wasn't used here. Removed dead code.